### PR TITLE
Qpcr work

### PR DESCRIPTION
**Pull request by matttgregg**
Work for the command based QPCR run, based on the existing plate reader workflow. Note that I've specified LHC component as input for future work, but it is currently passed through without change.
Comments welcome.

### CODE REVIEW

File: antha/anthalib/wtype/qpcrinstruction.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	ComponentIn        *LHComponent
+	ComponentOut   	   *LHComponent
+	Definition         string
+	Barcode 		   string
+	Command            string
+}
+
+func (ins QPCRInstruction) String() string {
+	return fmt.Sprint("QPCRInstruction")
+}
```
Comment (ddn0) [inline antha/anthalib/wtype/qpcrinstruction.go]: gofmt to remove these double newlines and to align field names

File: antha/anthalib/wtype/qpcrinstruction.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package wtype
+
+import (
+	"fmt"
+)
+
+
+// PRInstruction is a high-level instruction to a plate reader to measure a
```
Comment (ddn0) [inline antha/anthalib/wtype/qpcrinstruction.go]: QPCRInstruction ...

File: antha/anthalib/wtype/qpcrinstruction.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package wtype
```
Comment (ddn0) [inline antha/anthalib/wtype/qpcrinstruction.go]: should this be in wtype at all?

File: target/qpcrdevice/qpcrdevice.go [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+			TemplateFile: &quantstudio.ExperimentFile{
+				Url: inst.Definition,
+			},
+			Barcode: &quantstudio.Barcode{
+				Barcode: inst.Barcode,
+			},
+			OutputPath: "",
+		}
+
+		messageBytes, _ := proto.Marshal(message)
```
Comment (ddn0) [inline target/qpcrdevice/qpcrdevice.go]: probably should:
```
messageBytes, err := ...
if err != nil {
  return nil, err
}
```

File: antha/anthalib/wtype/qpcrinstruction.go (hunk 3 above)
Comment (matttgregg) [reply antha/anthalib/wtype/qpcrinstruction.go]: I'd agree with breaking that up - antha/ast or antha/antha/ast?

Comment (ddn0) [reply antha/anthalib/wtype/qpcrinstruction.go]: `antha/ast`.

`antha/antha/ast` is the AST for converting .an to .go, while, `antha/ast` is the AST for converting the execution of .go to a task graph.

File: execute/intrinsics.go [hunk 5]
```diff
@@ -278,12 +279,14 @@ func runQPCR(ctx context.Context, opts QPCROptions, command string) *commandInst
...
 }
 
+// RunQPCRExperiment starts a new QPCR experiment, using an experiment input file.
 func RunQPCRExperiment(ctx context.Context, opt QPCROptions) *wtype.LHComponent {
 	inst := runQPCR(ctx, opt, "RunExperiment")
 	trace.Issue(ctx, inst)
 	return inst.result[0]
 }
 
+// RunQPCRExperiment starts a new QPCR experiment, using a template input file.
```
Comment (ddn0) [inline execute/intrinsics.go]: `// RunQPCRFromTemplate`

**Pull request merged**